servers/quarkus-cli/src/main/java/org/projectnessie/quarkus/cli/ContentInfo.java

```diff
@@ -0,0 +1,301 @@
+      names = {"-H", "--hash"},
+      description = "Commit hash to use (defaults to the HEAD of the specified reference).")
+  private String hash;
+
+  @CommandLine.Option(
+      names = {"-s", "--summary"},
+      description = "Print a summary of results to STDOUT (irrespective of the --output option).")
+  private boolean summary;
+
+  private final AtomicInteger keysProcessed = new AtomicInteger();
+  private final AtomicInteger activeGlobalStateEntries = new AtomicInteger();
+  private final AtomicInteger missingContentFound = new AtomicInteger();
+
+  @Override
+  public Integer call() throws Exception {
+    warnOnInMemory();
+
+    if (outputSpec != null) {
+      if ("-".equals(outputSpec)) {
+        check(spec.commandLine().getOut());
+        spec.commandLine().getOut().println();
+      } else {
+        try (PrintWriter out = new PrintWriter(outputSpec, UTF_8)) {
+          check(out);
+        }
+      }
+    } else {
+      check(new PrintWriter(OutputStream.nullOutputStream(), false, UTF_8));
+    }
+
+    if (summary) {
+      spec.commandLine()
+          .getOut()
+          .printf(
+              "Processed %d keys: %d entries have global state; %d missing entries.%n",
+              keysProcessed.get(), activeGlobalStateEntries.get(), missingContentFound.get());
+    }
+
+    return missingContentFound.get() > 0 ? 3 : 0;
+  }
+
+  private void check(PrintWriter out) throws Exception {
+    // Note: Do not use try-with-resources to avoid closing the output. The caller takes care of
+    // that.
+    JsonGenerator generator =
+        new ObjectMapper()
+            .configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false)
+            .getFactory()
+            .createGenerator(out);
+
+    generator.writeStartArray();
+
+    check(generator);
+
+    generator.writeEndArray();
+    generator.flush();
+  }
+
+  private void check(JsonGenerator generator) throws Exception {
+    try (Stream<ReferenceInfo<ByteString>> heads = heads()) {
+      heads.forEach(
+          head -> {
+            try {
+              check(generator, head);
+            } catch (Exception e) {
+              throw new IllegalStateException(e);
+            }
+          });
+    }
+  }
+
+  private void check(JsonGenerator generator, ReferenceInfo<ByteString> head) throws Exception {
+    if (keyElements != null && !keyElements.isEmpty()) {
+      check(head, Collections.singleton(Key.of(keyElements)), generator);
+    } else {
+      Set<Key> batch = new HashSet<>(batchSize);
+      try (Stream<KeyListEntry> keys =
+          databaseAdapter.keys(head.getHash(), KeyFilterPredicate.ALLOW_ALL)) {
+        keys.forEach(
+            keyListEntry -> {
+              batch.add(keyListEntry.getKey());
+
+              if (batch.size() >= batchSize) {
+                check(head, batch, generator);
+                batch.clear();
+              }
+            });
+      }
+
+      check(head, batch, generator); // check remaining keys
```

Comment:
Should be guarded with a `if (!batch.isEmpty())`

Reply:
good catch - fixed